python/lsst/ts/observatory/control/maintel/comcam.py

```diff
@@ -240,7 +241,43 @@ async def setup_filter(self, filter):
         if filter is None:
             return
         else:
-            raise NotImplementedError("set_filter not available yet.")
+            self.rem.cccamera.cmd_setFilter.set(name=filter)
+            async with self.cmd_lock:
+                self.rem.cccamera.evt_endSetFilter.flush()
+                await self.rem.cccamera.cmd_setFilter.start(
+                    timeout=self.filter_change_timeout
+                )
+                end_setFilter = await self.rem.cccamera.evt_endSetFilter.next(
+                    flush=False, timeout=self.filter_change_timeout
+                )
+                self.log.info(f"Filter {end_setFilter.filterName} in position.")
+                return end_setFilter
+
+    async def get_current_filter(self):
+        """Get the current filter.
+
+        Returns
+        -------
+        `str`
+            The filter in the light path.
+        """
+        end_setFilter = await self.rem.cccamera.evt_endSetFilter.aget(
```

Comment:
May be a good idea to do some error handling here. You could encapsulate the await call with try/except, catch a `TimeoutError` and raise a more user-friendly message. Most users get confused with `TimeoutError` in methods like this. 

Something like this:

```
try:
    end_setFilter = await self.rem.cccamera.evt_endSetFilter.aget(...)
except asyncio.TimeoutError:
    raise RuntimeError("Could not determine current filter. Either data was never published or historical data is not working properly.")
```

Reply:
Done.